transport/shadowsocks/packet_listener.go

```diff
@@ -65,6 +64,10 @@ type packetConn struct {
 
 var _ net.PacketConn = (*packetConn)(nil)
 
+func NewPacketConn(conn net.Conn, key *EncryptionKey) net.PacketConn {
```

Comment:
Please add a comment.

Is the inner conn closed on the outer conn close? May be worth commenting.

Reply:
Done.